src/actions/actions.ts

```diff
@@ -4980,7 +4990,35 @@ class ActionDeleteLineVisualMode extends BaseCommand {
   keys = ["X"];
 
   public async exec(position: Position, vimState: VimState): Promise<VimState> {
-    return await new DeleteOperator(this.multicursorIndex).run(vimState, position.getLineBegin(), position.getLineEnd());
+    if (vimState.currentMode === ModeName.Visual) {
+      return await new DeleteOperator(this.multicursorIndex).run(vimState,
+                                                                vimState.lastVisualSelectionStart.getLineBegin(),
```

Comment:
Why is it `lastVisualSelection`? Is the current selection not what we want?

`lastSelection` was intended for commands like gv that are triggered after you stopped selecting entirely and did something else, so I just wanna make sure this is correct.